language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/hover/HoverPerformanceTest.java

```diff
@@ -25,31 +25,44 @@
 
 import java.io.IOException;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Test performance of hover feature in language server.
  *
  * @since 2.0.0
  */
 public class HoverPerformanceTest extends HoverProviderTest {
-    private JsonParser parser = new JsonParser();
+
+    private static final int TEST_ITERATIONS = 10;
+    
+    private final List<Long> executionTimes = new ArrayList<>();
 
     @Test(description = "Test Hover provider", dataProvider = "hover-data-provider")
     public void testHover(String config) throws IOException {
-        super.testHover(config);
+        // We run the same test multiple times and take the average of them as the execution time. This is to
+        // reduce the effect of first compilation and outliers
+        for (int i=0; i<TEST_ITERATIONS; i++) {
+            super.testHover(config);
+        }
```

Comment:
Here we do `TestUtil.openDocument(serviceEndpoint, sourcePath)` and `TestUtil.closeDocument(serviceEndpoint, sourcePath)` 10 times right? So compilation happens 10 times

Reply:
Makes sense.That didn't come to my mind while doing the change. However, taking average over several runs is better right? Even if the compilation is still happening?

Reply:
Yeah taking an average makes sense, what do you think of getting the average only for `getHoverResponse` calls? (Since openDocument is costly and we are not taking any use of those later openDocument calls). 

And is there a specific reason for doing this only for hoverPerformanceTest?